pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -57,14 +57,20 @@ type AdvertisableAddressResolver interface {
 	Resolve(observedAdddress ma.Multiaddr) (ma.Multiaddr, error)
 }
 
+// GuardedMessage is a string message guarded by a RW mutex.
+type GuardedMessage struct {
```

Comment:
I think there is no need for Guarded in the name here.  Also, instead of a struct, you could have also just added `messageMu` field to service and use it to guard a message, so there would be no need for additional struct, especially since methods that actually use the mutex are on service level and not on `GuarderMessage` receiver.

Reply:
If you choose to leave the struct, I would ommit the `Guarded` in the name, and I would make get and set methood on a `GuardedMessage` receiver.

Reply:
you can use `atomic.Value` for [this](https://golang.org/pkg/sync/atomic/#Value).